<!--/*****************************************************************
 * labust_primitives_template.launch
 *
 *  Created on: Apr 16, 2014
 *      Author: Filip Mandic
 *
 ********************************************************************/

/*********************************************************************
* Software License Agreement (BSD License)
*
*  Copyright (c) 2014, LABUST, UNIZG-FER
*  All rights reserved.
*
*  Redistribution and use in source and binary forms, with or without
*  modification, are permitted provided that the following conditions
*  are met:
*
*   * Redistributions of source code must retain the above copyright
*     notice, this list of conditions and the following disclaimer.
*   * Redistributions in binary form must reproduce the above
*     copyright notice, this list of conditions and the following
*     disclaimer in the documentation and/or other materials provided
*     with the distribution.
*   * Neither the name of the LABUST nor the names of its
*     contributors may be used to endorse or promote products derived
*     from this software without specific prior written permission.
*
*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
*  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
*  POSSIBILITY OF SUCH DAMAGE.
******************************************************************/-->

<launch>

    <!-- ========================================================= -->
    <!-- == Arguments ============================================ -->
    <!-- ========================================================= -->


    <!-- ========================================================= -->
    <!-- == CourseKeeping FA node ================================ -->
    <!-- ========================================================= -->

    <node 
        pkg="labust_primitives" 
        type="course_keeping_FA"
        name="course_keeping_FA" >

     	<remap from="out" to="stateRef" />
     	<remap from="state" to="stateHat" />
	<param name="underactuated" value="false" />

    </node>

    <!-- ========================================================= -->
    <!-- == CourseKeeping UA node ================================ -->
    <!-- ========================================================= -->

    <node 
        pkg="labust_primitives" 
        type="course_keeping_UA"
        name="course_keeping_UA" >

     	<remap from="out" to="stateRef" />
     	<remap from="state" to="stateHat" />
	<param name="underactuated" value="true" />

    </node>

    <!-- ========================================================= -->
    <!-- == Dynamic Positioning node ============================= -->
    <!-- ========================================================= -->

    <node 
        pkg="labust_primitives"
        type="dp_primitive"
        name="dp_primitive">
     	
        <remap from="out" to="stateRef" />
     	<remap from="state" to="stateHat" />
        <param name="underactuated" value="false" />

     </node>

    <!-- ========================================================= -->
    <!-- == Go2Point FA node ===================================== -->
    <!-- ========================================================= -->

    <node 
        pkg="labust_primitives"
        type="go2point_FA"
        name="go2point_FA">

     	<remap from="out" to="stateRef" />
     	<remap from="state" to="stateHat" />
        <param name="underactuated" value="false" />

    </node>

    <!-- ========================================================= -->
    <!-- == Go2Point UA node ===================================== -->
    <!-- ========================================================= -->

    <node 
        pkg="labust_primitives"
        type="go2point_UA"
        name="go2point_UA">

     	<remap from="out" to="stateRef" />
     	<remap from="state" to="stateHat" />
        <param name="underactuated" value="true" />

    </node>
   
</launch>
